### PR TITLE
fix(sol): fix deserializing signed sol transaction

### PR DESCRIPTION
**Pull request by joey-bai**
when deserializing a sol transaction, we would lose out on durable nonce
info. this fixes it.

Ticket: STLX-10065

### CODE REVIEW

File: modules/account-lib/src/coin/sol/transaction.ts [hunk 1]
```diff
@@ -140,10 +140,21 @@ export class Transaction extends BaseTransaction {
...
+        walletNonceAddress: nonceInstruction.noncePubkey.toString(),
+        authWalletAddress: nonceInstruction.authorizedPubkey.toString(),
+      };
+    }
+
     const result: TxData = {
       id: this.id,
       feePayer: this._solTransaction.feePayer?.toString(),
       nonce: this.getNonce(),
+      durableNonce: durableNonce,
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/sol/transaction.ts:157]: So here in TxData durable nonce could be undefined?
We should build a TxData with and without this field?
@CapnMigraine , @joey-bai what you recommend here?

Comment (alebusse) [reply modules/account-lib/src/coin/sol/transaction.ts:157]: Undefined should be fine. Idk whats gonna be this field data be used for

**Review by alebusse — APPROVED** [review]

File: modules/account-lib/src/coin/sol/transaction.ts (hunk 1 above)
Comment (joey-bai) [reply modules/account-lib/src/coin/sol/transaction.ts:157]: @rliberal-simtlix yes, the blockhash/nonce can come from two sources:
- recent value from full node
- value from a durable nonce account

`durableNonce` is used to indicate that the nonce value comes from the nonce account.
if this field is empty, then the nonce is a recent value from the full node.

Comment (rliberal-simtlix) [reply modules/account-lib/src/coin/sol/transaction.ts:158]: Sounds good. My question was related in order to add increase the payload size of tx, which base on the fee the tx per byte could increase cost. But in that case that you mention i think is ok.
Thanks for the clarification.

**Review by rliberal-simtlix — APPROVED** [review]

**Pull request merged**